### ISSUE TITLE
`:config-in-ns` is not used when ignoring `:discouraged-namespace` linter

### ISSUE THREAD

**Issue by eval-on-point** [clj-kondo/clj-kondo#2424]
- [x] I have read the [Clojure etiquette](https://clojure.org/community/etiquette) and will respect it when communicating on this platform.

**version**

```shell
$ clj-kondo --version
clj-kondo v2024.09.27
```

**platform**

native image (from nixpkgs-unstable)

**problem**

`clj-kondo` does not take the `:config-in-ns` configuration into account when ignoring the `:discouraged-namespace` linter. For example, the below `repro.clj` and `config.edn` should not produce the `discouraged ns` warning.

```shell
$ clj-kondo --lint repro.clj
repro.clj:2:14: warning: namespace discouraged is required but never used
repro.clj:2:14: warning: discouraged ns
linting took 7ms, errors: 0, warnings: 2
```

**repro**

**repro.clj**
```clj
(ns repro
  (:require [discouraged :as d]))
```

**config**

```edn
{:config-in-ns {repro {:ignore [:discouraged-namespace]}}
 :linters {:discouraged-namespace {discouraged {:message "discouraged ns"}}}}
```

**expected behavior**

```sh
$ clj-kondo --lint repro.clj
repro.clj:2:14: warning: namespace discouraged is required but never used
linting took 7ms, errors: 0, warnings: 1
```
